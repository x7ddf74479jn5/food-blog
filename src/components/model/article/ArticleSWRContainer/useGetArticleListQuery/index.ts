import type { MicroCMSQueries } from "microcms-js-sdk";
import { useCallback, useMemo, useRef } from "react";
import useSWRInfinite from "swr/infinite";

import type { TArticleListResponse, TArticleSWRResponse } from "@/types";
import { HttpError } from "@/utils/error/Http";
import { apiRoute } from "@/utils/paths/url";

const createURLSearchParams = (data: Record<string, string | number>) => {
  const params = new URLSearchParams();
  Object.entries(data).forEach(([key, value]) => {
    const v = typeof value === "string" ? value : value.toString();
    params.append(key, v);
  });

  return params.toString();
};

type Arguments = {
  queries?: Omit<MicroCMSQueries, "fields" | "ids">;
  fallbackData?: TArticleListResponse;
};

const useGetArticleListQuery = ({ fallbackData, queries }: Arguments) => {
  const endpoint = apiRoute.apiArticles;
  const keyRef = useRef("");
  const defaultLimit = 10;
  const defaultKeyOptions = {
    limit: defaultLimit,
  };
  const defaultFetcher = (url: string): Promise<TArticleListResponse> =>
    fetch(url).then((res) => {
      if (!res.ok) {
        throw new HttpError(res);
      }
      return res.json();
    });

  const getKey = (pageIndex: number, previousPageData: TArticleSWRResponse) => {
    if (previousPageData && !previousPageData.contents) return null;

    if (pageIndex === 0) {
      const key = `${endpoint}?${createURLSearchParams({ ...defaultKeyOptions, ...queries, pageIndex })}`;
      keyRef.current = key;

      return key;
    }

    const offset =
      previousPageData?.offset !== undefined ? previousPageData.offset + (queries?.limit ?? defaultLimit) : 0;
    const key = `${endpoint}?${createURLSearchParams({ ...defaultKeyOptions, ...queries, offset, pageIndex })}`;
    keyRef.current = key;

    return key;
  };

  const result = useSWRInfinite<TArticleSWRResponse, Error>(getKey, defaultFetcher, {
    fallbackData: fallbackData ? [fallbackData] : undefined,
    revalidateIfStale: false,
    revalidateOnFocus: false,
    revalidateOnReconnect: false,
  });

  const { data, mutate, setSize, size, ...rest } = result;

  const getCurrentKey = useCallback(() => {
    return keyRef.current;
  }, [keyRef]);

  const revalidate = useCallback(() => {
    mutate();
  }, [mutate]);

  const paginate = useCallback(() => {
    setSize(size + 1);
  }, [setSize, size]);

  const articles = useMemo(() => {
    return data ? data.map((r) => (r ? r.contents : [])).flat() : [];
  }, [data]);

  const totalCount = useMemo(() => {
    return data ? data[0]?.totalCount : 0;
  }, [data]);

  const hasNextPage = useMemo(() => {
    return totalCount ? articles.length !== totalCount : false;
  }, [articles.length, totalCount]);

  return { ...rest, articles, getCurrentKey, hasNextPage, paginate, revalidate } as const;
};

export default useGetArticleListQuery;
