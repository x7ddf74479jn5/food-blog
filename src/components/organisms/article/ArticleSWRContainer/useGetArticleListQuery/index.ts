import type { MicroCMSQueries } from "microcms-js-sdk";
import { useCallback, useMemo, useRef } from "react";
import useSWRInfinite from "swr/infinite";

import type { TArticleListResponse, TArticleSWRResponse } from "@/types";
import { HttpError } from "@/utils/error/Http";
import { apiRoute } from "@/utils/paths/url";

const createURLSearchParams = (data: Record<string, string | number>) => {
  const params = new URLSearchParams();
  Object.entries(data).forEach(([key, value]) => {
    const v = typeof value === "string" ? value : value.toString();
    params.append(key, v);
  });

  return params.toString();
};

type Arguments = {
  getKeyOptions?: Omit<MicroCMSQueries, "fields" | "ids">;
  fallbackData?: TArticleListResponse;
};

const useGetArticleListQuery = ({ fallbackData, getKeyOptions }: Arguments) => {
  const keyRef = useRef("");
  const endpoint = apiRoute.apiArticles;
  const defaultLimit = 10;
  const defaultKeyOptions = {
    limit: defaultLimit,
  };
  const defaultFetcher = (url: string): Promise<TArticleListResponse> =>
    fetch(url).then((res) => {
      if (!res.ok) {
        throw new HttpError(res);
      }
      return res.json();
    });

  const getKey = (pageIndex: number, previousPageData: TArticleSWRResponse) => {
    if (previousPageData && !previousPageData.contents) return null;

    if (pageIndex === 0) {
      const key = `${endpoint}?${createURLSearchParams({ ...defaultKeyOptions, ...getKeyOptions, pageIndex })}`;
      keyRef.current = key;

      return key;
    }

    const offset =
      previousPageData?.offset !== undefined ? previousPageData.offset + (getKeyOptions?.limit ?? defaultLimit) : 0;
    const key = `${endpoint}?${createURLSearchParams({ ...defaultKeyOptions, ...getKeyOptions, offset, pageIndex })}`;
    keyRef.current = key;

    return key;
  };

  const result = useSWRInfinite<TArticleSWRResponse, Error>(getKey, defaultFetcher, {
    fallbackData: fallbackData ? [fallbackData] : undefined,
    revalidateOnFocus: false,
    revalidateOnReconnect: false,
  });

  const { data, mutate, setSize, size, ...rest } = result;

  const getCurrentKey = useCallback(() => {
    return keyRef.current;
  }, [keyRef]);

  const revalidate = useCallback(() => {
    mutate();
  }, [mutate]);

  const paginate = useCallback(() => {
    setSize(size + 1);
  }, [setSize, size]);

  const articles = useMemo(() => {
    return data?.flatMap((r) => r?.contents ?? []) ?? [];
  }, [data]);

  const totalCount = useMemo(() => {
    return data ? data[0]?.totalCount : 0;
  }, [data]);

  const hasNextPage = useMemo(() => {
    return totalCount ? articles.length !== totalCount : false;
  }, [articles.length, totalCount]);

  return { ...rest, articles, getCurrentKey, hasNextPage, paginate, revalidate } as const;
};

export default useGetArticleListQuery;
