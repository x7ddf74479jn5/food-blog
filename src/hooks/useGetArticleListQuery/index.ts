import { useCallback, useMemo, useRef } from "react";
import useSWRInfinite from "swr/infinite";

import type { TApiRoute, TArticleListResponse, TArticleSWRResponse, TQueryOptions } from "@/types";
import { HttpError } from "@/utils/error/Http";

const createURLSearchParams = (data: Record<string, string | number>) => {
  const params = new URLSearchParams();
  Object.entries(data).forEach(([key, value]) => {
    const v = typeof value === "string" ? value : value.toString();
    params.append(key, v);
  });

  return params.toString();
};

type Arguments = {
  endpoint: TApiRoute;
  getKeyOptions?: TQueryOptions;
  fallbackData?: TArticleListResponse;
};

const useGetArticleListQuery = ({ endpoint, getKeyOptions, fallbackData }: Arguments) => {
  const keyRef = useRef("");
  const defaultLimit = 10;
  const defaultKeyOptions = {
    limit: defaultLimit,
  };
  const defaultFetcher = (url: string): Promise<TArticleListResponse> =>
    fetch(url).then((res) => {
      if (!res.ok) {
        throw new HttpError(res);
      }
      return res.json();
    });

  const getKey = (pageIndex: number, previousPageData: TArticleSWRResponse) => {
    if (previousPageData && !previousPageData.contents) return null;

    if (pageIndex === 0) {
      const key = `${endpoint}?${createURLSearchParams({ ...defaultKeyOptions, ...getKeyOptions, pageIndex })}`;
      keyRef.current = key;

      return key;
    }

    const offset =
      previousPageData?.offset !== undefined ? previousPageData.offset + (getKeyOptions?.limit ?? defaultLimit) : 0;
    const key = `${endpoint}?${createURLSearchParams({ ...defaultKeyOptions, ...getKeyOptions, offset, pageIndex })}`;
    keyRef.current = key;

    return key;
  };

  const result = useSWRInfinite<TArticleSWRResponse, Error>(getKey, defaultFetcher, {
    revalidateIfStale: false,
    revalidateOnFocus: false,
    revalidateOnReconnect: false,
    fallbackData: fallbackData ? [fallbackData] : undefined,
  });

  const { size, setSize, mutate, data, ...rest } = result;

  const getCurrentKey = useCallback(() => {
    return keyRef.current;
  }, [keyRef]);

  const revalidate = useCallback(() => {
    mutate();
  }, [mutate]);

  const paginate = useCallback(() => {
    setSize(size + 1);
  }, [setSize, size]);

  const articles = useMemo(() => {
    return data ? data.map((r) => (r ? r.contents : [])).flat() : [];
  }, [data]);

  const totalCount = useMemo(() => {
    return data ? data[0]?.totalCount : 0;
  }, [data]);

  const hasNextPage = useMemo(() => {
    return totalCount ? articles.length !== totalCount : false;
  }, [articles.length, totalCount]);

  return { ...rest, getCurrentKey, revalidate, paginate, hasNextPage, articles } as const;
};

export default useGetArticleListQuery;
