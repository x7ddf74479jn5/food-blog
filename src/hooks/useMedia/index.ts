import { useEffect, useState } from "react";

const breakpoints = {
  xs: { min: 0, max: 639 },
  sm: { min: 640, max: 767 },
  md: { min: 768, max: 1023 },
  lg: { min: 1024, max: 1279 },
  xl: { min: 1280, max: 1535 },
  "2xl": { min: 1536, max: 999999 },
};

type Size = keyof typeof breakpoints;
type MatchType = ">=" | "<=" | "=";

/**
 * example:
 *
 * const isSmallOrDown = useMedia("(max-width: 767px)")
 * const isMedium = useMedia("(min-width: 768px) and (max-width: 1023px)")
 * const isLargeOrUp = useMedia("(min-width: 1024px)")
 */
export const useMediaQuery = (query: string) => {
  const [isMatch, setMatch] = useState(typeof window === "undefined" ? false : window.matchMedia(query).matches);

  useEffect(() => {
    const media = window.matchMedia(query);
    if (media.matches !== isMatch) {
      setMatch(media.matches);
    }

    const listener = () => setMatch(media.matches);
    /**
     * addListener は非推奨だが、古いブラウザのために使用している。
     * 古いブラウザを切るなら window.addEventListener("resize", listener) が推奨。
     */
    media.addListener(listener);
    return () => media.removeListener(listener);

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [query]);

  return isMatch;
};

const getQuery = (matchType: MatchType, size: Size) => {
  if (matchType === ">=") {
    return `(min-width: ${breakpoints[size].min}px)`;
  } else if (matchType === "<=") {
    return `(max-width: ${breakpoints[size].max}px)`;
  } else {
    return `(min-width: ${breakpoints[size].min}px) and (max-width: ${breakpoints[size].max}px)`;
  }
};

/**
 * example:
 *
 * const isSmallOrDown = useMedia("<=", "sm") // (max-width: 767px)
 * const isMedium = useMedia("=", "md") // (min-width: 768px) and (max-width: 1023px)
 * const isLargeOrUp = useMedia(">=", "lg") // (min-width: 1024px)
 */
export const useMedia = (matchType: MatchType, size: Size) => {
  return useMediaQuery(getQuery(matchType, size));
};
