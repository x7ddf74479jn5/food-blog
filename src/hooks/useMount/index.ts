import { useEffect, useState } from "react";

export const useMount = () => {
  const [isMounted, setMounted] = useState(false);

  useEffect(() => {
    if (!isMounted) setMounted(true);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return isMounted;
};
