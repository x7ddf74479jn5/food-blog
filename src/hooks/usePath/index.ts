import { usePathname } from "next/navigation";
import { useCallback } from "react";

export const usePath = () => {
  const pathname = usePathname();
  const matchPath = useCallback((path: string) => pathname === path, [pathname]);

  return { matchPath };
};
