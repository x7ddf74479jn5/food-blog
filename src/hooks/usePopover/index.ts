import { useCallback, useEffect, useRef, useState } from "react";

export const usePopover = <E extends HTMLElement>() => {
  const ref = useRef<E>(null);
  const [isActive, _setIsActive] = useState(false);

  useEffect(() => {
    const handleClickOutside = (event: Event) => {
      if (ref.current && !ref.current.contains(event.target as Node)) {
        _setIsActive(false);
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  });

  const setIsActive = useCallback((flag: boolean) => _setIsActive(flag), []);

  return {
    ref,
    isActive,
    setIsActive,
  };
};
