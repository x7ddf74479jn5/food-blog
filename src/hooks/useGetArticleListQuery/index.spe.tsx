import { renderHook } from "@testing-library/react-hooks";
import { withMockedRouter } from "jest/mocks/router";
import { server } from "mocks/msw/server";
import type { NextRouter } from "next/router";
import { SWRConfig } from "swr";

import useGetArticleListQuery from ".";

beforeAll(() => server.listen());
afterAll(() => server.close());

describe("hooks/useGetArticleListQuery", () => {
  it("should ", () => {
    const router: Partial<NextRouter> = {
      query: { q: "" },
    };

    const Wrapper: React.ComponentType<{ children: React.ReactNode; router: Partial<NextRouter> }> = ({ children }) => {
      return withMockedRouter(
        router,
        <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>{children}</SWRConfig>
      );
    };

    // const make = (router: Partial<NextRouter>) => wrapper({ router });

    const result = renderHook(() => useGetArticleListQuery({}), { wrapper: Wrapper }).result;
    expect(result);
  });
});
