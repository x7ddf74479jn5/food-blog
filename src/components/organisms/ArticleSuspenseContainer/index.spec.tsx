import { mockArticles } from "@mocks/data";
import { fireEvent, render, screen } from "jest/test-utils";
import renderer from "react-test-renderer";

import * as useGetArticleQuery from "@/hooks/useGetArticleListQuery";

import { ArticleSuspenseContainer } from ".";

let spyUseGetArticleQuery: jest.SpyInstance;
const mockPaginate = jest.fn();

describe("components/molecules/ArticleSuspenseContainer", () => {
  describe("OK: データがある", () => {
    describe("次ページがある", () => {
      beforeEach(() => {
        spyUseGetArticleQuery = jest.spyOn(useGetArticleQuery, "default");
        spyUseGetArticleQuery.mockReturnValue({
          articles: [mockArticles.stock],
          hasNextPage: true,
          error: undefined,
          isValidating: false,
          paginate: mockPaginate,
          setSize: jest.fn(),
          size: 1,
          data: [],
          getCurrentKey: jest.fn(),
          revalidate: jest.fn(),
          mutate: jest.fn(),
        });
      });
      afterEach(() => {
        jest.clearAllMocks();
        jest.restoreAllMocks();
      });

      it("Snapshot", () => {
        const tree = renderer.create(<ArticleSuspenseContainer />).toJSON();
        expect(tree).toMatchSnapshot();
      });

      it("OK: 初期レンダリング", () => {
        render(<ArticleSuspenseContainer />);
        const article = screen.getByRole("article");
        expect(article).toBeInTheDocument();
        const button = screen.getByRole("button");
        expect(button).toBeInTheDocument();
      });

      it("Interaction", () => {
        render(<ArticleSuspenseContainer />);
        const button = screen.getByRole("button");
        expect(button).toBeInTheDocument();
        expect(button).toBeEnabled();
        fireEvent.click(button);
        expect(mockPaginate).toBeCalled();
      });
    });

    describe("次ページがない", () => {
      beforeEach(() => {
        spyUseGetArticleQuery = jest.spyOn(useGetArticleQuery, "default");
        spyUseGetArticleQuery.mockReturnValue({
          articles: [mockArticles.stock],
          hasNextPage: false,
          error: undefined,
          isValidating: false,
          paginate: mockPaginate,
          setSize: jest.fn(),
          size: 1,
          data: [],
          getCurrentKey: jest.fn(),
          revalidate: jest.fn(),
          mutate: jest.fn(),
        });
      });
      afterEach(() => {
        jest.clearAllMocks();
        jest.restoreAllMocks();
      });
      it("Snapshot", () => {
        const tree = renderer.create(<ArticleSuspenseContainer />).toJSON();
        expect(tree).toMatchSnapshot();
      });

      it("OK: 初期レンダリング", () => {
        render(<ArticleSuspenseContainer />);
        const article = screen.getByRole("article");
        expect(article).toBeInTheDocument();
        const button = screen.queryByRole("button");
        expect(button).not.toBeInTheDocument();
      });
    });

    describe("ローディング", () => {
      beforeEach(() => {
        spyUseGetArticleQuery = jest.spyOn(useGetArticleQuery, "default");
        spyUseGetArticleQuery.mockReturnValue({
          articles: [mockArticles.stock],
          hasNextPage: true,
          error: undefined,
          isValidating: true,
          paginate: mockPaginate,
          setSize: jest.fn(),
          size: 1,
          data: [],
          getCurrentKey: jest.fn(),
          revalidate: jest.fn(),
          mutate: jest.fn(),
        });
      });
      afterEach(() => {
        jest.clearAllMocks();
        jest.restoreAllMocks();
      });

      it("Snapshot", () => {
        const tree = renderer.create(<ArticleSuspenseContainer />).toJSON();
        expect(tree).toMatchSnapshot();
      });

      it("OK: 初期レンダリング", () => {
        render(<ArticleSuspenseContainer />);
        const button = screen.queryByRole("button");
        expect(button).not.toBeInTheDocument();
      });
    });
  });

  describe("OK: データがない", () => {
    beforeEach(() => {
      spyUseGetArticleQuery = jest.spyOn(useGetArticleQuery, "default");
      spyUseGetArticleQuery.mockReturnValue({
        articles: [],
        hasNextPage: false,
        error: undefined,
        isValidating: false,
        paginate: mockPaginate,
        setSize: jest.fn(),
        size: 1,
        data: [],
        getCurrentKey: jest.fn(),
        revalidate: jest.fn(),
        mutate: jest.fn(),
      });
    });

    it("Snapshot", () => {
      const tree = renderer.create(<ArticleSuspenseContainer />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("OK: 初期レンダリング", () => {
      const { container } = render(<ArticleSuspenseContainer />);

      const article = screen.queryByRole("article");
      expect(article).not.toBeInTheDocument();
      const button = screen.queryByRole("button");
      expect(button).not.toBeInTheDocument();
      expect(container).toHaveTextContent("レシピが見つかりませんでした。");
    });
  });

  describe("NG:  エラー", () => {
    beforeEach(() => {
      spyUseGetArticleQuery = jest.spyOn(useGetArticleQuery, "default");
      spyUseGetArticleQuery.mockReturnValue({
        articles: [],
        hasNextPage: false,
        error: new Error(),
        isValidating: false,
        paginate: mockPaginate,
        setSize: jest.fn(),
        size: 1,
        data: [],
        getCurrentKey: jest.fn(),
        revalidate: jest.fn(),
        mutate: jest.fn(),
      });
    });

    it("Snapshot", () => {
      const tree = renderer.create(<ArticleSuspenseContainer />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("OK: 初期レンダリング", () => {
      const { container } = render(<ArticleSuspenseContainer />);

      const article = screen.queryByRole("article");
      expect(article).not.toBeInTheDocument();
      const button = screen.queryByRole("button");
      expect(button).not.toBeInTheDocument();
      expect(container).toHaveTextContent("エラーが発生しました。");
    });
  });
});
