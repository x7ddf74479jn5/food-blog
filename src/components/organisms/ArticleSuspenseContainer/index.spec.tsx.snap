// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/ArticleSuspenseContainer OK: データがある ローディング Snapshot 1`] = `
<section>
  <ul
    className="space-y-12"
  >
    <li>
      <article>
        <div
          className="mb-4"
        >
          <a
            aria-label="基本の一番だしの作り方"
            href="/articles/1"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI4MCIgaGVpZ2h0PSI3MjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="Cover Image for 基本の一番だしの作り方"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="Cover Image for 基本の一番だしの作り方"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=3840&q=75"
                  srcSet="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2F5026416_s.jpg&w=3840&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </a>
        </div>
        <h2
          className="mb-4 text-2xl font-bold"
        >
          <a
            href="/articles/1"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            基本の一番だしの作り方
          </a>
        </h2>
        <p
          className="dark:text-gray-300 line-clamp-2"
        >
          だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。
        </p>
      </article>
    </li>
  </ul>
  <div
    className="flex justify-center mt-8"
  >
    <div
      className="w-8 h-8 rounded-full border-b-2 border-green-500 animate-spin"
    />
  </div>
</section>
`;

exports[`components/molecules/ArticleSuspenseContainer OK: データがある 次ページがある Snapshot 1`] = `
<section>
  <ul
    className="space-y-12"
  >
    <li>
      <article>
        <div
          className="mb-4"
        >
          <a
            aria-label="基本の一番だしの作り方"
            href="/articles/1"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI4MCIgaGVpZ2h0PSI3MjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="Cover Image for 基本の一番だしの作り方"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="Cover Image for 基本の一番だしの作り方"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=3840&q=75"
                  srcSet="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2F5026416_s.jpg&w=3840&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </a>
        </div>
        <h2
          className="mb-4 text-2xl font-bold"
        >
          <a
            href="/articles/1"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            基本の一番だしの作り方
          </a>
        </h2>
        <p
          className="dark:text-gray-300 line-clamp-2"
        >
          だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。
        </p>
      </article>
    </li>
  </ul>
  <div
    className="flex justify-center mt-8"
  >
    <button
      className="flex flex-row gap-2 justify-center items-center py-2 px-4 text-white bg-green-700 hover:bg-opacity-90 rounded-xl border-2 border-green-600"
      onClick={[MockFunction]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        />
      </svg>
      <span>
        もっと読み込む
      </span>
    </button>
  </div>
</section>
`;

exports[`components/molecules/ArticleSuspenseContainer OK: データがある 次ページがない Snapshot 1`] = `
<section>
  <ul
    className="space-y-12"
  >
    <li>
      <article>
        <div
          className="mb-4"
        >
          <a
            aria-label="基本の一番だしの作り方"
            href="/articles/1"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI4MCIgaGVpZ2h0PSI3MjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="Cover Image for 基本の一番だしの作り方"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="Cover Image for 基本の一番だしの作り方"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=3840&q=75"
                  srcSet="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2F5026416_s.jpg&w=3840&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </a>
        </div>
        <h2
          className="mb-4 text-2xl font-bold"
        >
          <a
            href="/articles/1"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            基本の一番だしの作り方
          </a>
        </h2>
        <p
          className="dark:text-gray-300 line-clamp-2"
        >
          だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。
        </p>
      </article>
    </li>
  </ul>
</section>
`;

exports[`components/molecules/ArticleSuspenseContainer OK: データがない Snapshot 1`] = `
<div
  className="flex justify-center mt-16"
>
  レシピが見つかりませんでした。
</div>
`;
